src/dbnode/namespace/convert.go

```diff
@@ -186,6 +186,11 @@ func ToMetadata(
 		return nil, err
 	}
 
+	stagingState, err := ToStagingState(opts.StagingState)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I think this breaks back compat; if receiving old messages, the staging set will always be `UNKNOWN`, which will error; UNKNOWN should probably be ignored here instead

Reply:
Unless I'm missing something, old messages will have a nil `state` which would end up getting back `nil` from `ToStagingState` which is what we want. Coordinator code will know that nil staging state means the namespace was created before we added the option and can act accordingly.

Reply:
Ah I see, looks like `opts.StagingState` is nilable so we won't get the default values here, but we should still be careful around not using `.GetStatus()` instead of `.Status` https://github.com/m3db/m3/pull/2721/files#diff-4778e9197fdf5ce965022ca643eb03c2R416-R420

IMO `UNKNOWN` should just be identical to having this be a legacy state and ignored, wdyt?

Reply:
Ah, interesting point re: `.GetStatus()`. Yeah, we can treat `UNKNOWN` as legacy pretty easily. Will add that.